app.py

```diff
@@ -152,23 +198,37 @@ def parse():
         'server_error': response.get('Error'),
         'version': version(),
         'username': username
-    })
+    }
+
+    if os.getenv ('REDIRECT_AB_TEST') == os.getenv ('HEROKU_APP_NAME') and os.getenv ('HEROKU_APP_NAME'):
+        log_object ['is_test'] = 1
+        if 'x-session_id' in request.headers:
+            log_object ['session'] = request.headers ['x-session_id']
+
+    logger.log(log_object)
 
     return jsonify(response)
 
 @app.route('/report_error', methods=['POST'])
 def report_error():
     post_body = request.json
 
-    logger.log({
+    log_object = {
         'session': session_id(),
         'date': str(datetime.datetime.now()),
         'level': post_body.get('level'),
         'code': post_body.get('code'),
         'client_error': post_body.get('client_error'),
         'version': version(),
         'username': current_user(request) ['username'] or None
-    })
+    }
+
+    if os.getenv ('REDIRECT_AB_TEST') == os.getenv ('HEROKU_APP_NAME') and os.getenv ('HEROKU_APP_NAME'):
```

Comment:
This is duplicated with the previous block.

Should have been put into a separate function for deduplication purposes, or go into the `pre_request` trick.